src/wp-includes/global-styles-and-settings.php

```diff
@@ -264,6 +264,23 @@ function ( $item ) {
  * @return bool Returns true if theme or its parent has a theme.json file, false otherwise.
  */
 function wp_theme_has_theme_json() {
+	static $theme_has_support = null;
+
+	/*
+	 * Ignore cache when `WP_DEBUG` is enabled. Why?
+	 * - to avoid interfering with the theme developer's workflow.
+	 * - to reset between each automated test.
+	 *
+	 * @todo Replace `WP_DEBUG` once an "in development mode" check is available in Core.
+	 */
+	if ( WP_DEBUG ) {
+		$theme_has_support = null;
+	}
```

Comment:
As noted in https://github.com/WordPress/wordpress-develop/pull/3887#issuecomment-1400875992, a static cache reset is needed to reset after each automated test. This can be accomplished in many ways:

- Checking constants like `WP_DEBUG` and/or `WP_RUN_CORE_TESTS`
- Adding a hard reset parameter to the function
- Filtering
- Global

There is precedence within Core for using the `WP_DEBUG` and `WP_RUN_CORE_TESTS constants.

I'd suggest avoiding the approach of adding a reset param to the function. Why? (a) BC concerns for the future if the param is no longer needed and (b) performance concerns of exposing a hard reset in prod.

However if a hard reset option is needed, for example on theme switch or preview, then the param can be added and constant(s) removed.

cc @felixarntz @azaozz @oandregal What do you think?

Reply:
Could also combine the check with early bail-out below.

Reply:
@hellofromtonya +1 to avoid the function parameter, I'd be concerned about BC with that too.

IMO the `WP_DEBUG` approach is the most suitable for now, since that's what we were going to go with at this point anyway (keeping in consideration the `@todo` comment to iterate on it with a more appropriate solution). Later, once there is e.g. a filter for this too, we would be able to adjust the tests themselves to decide on whether or not the static variable cache should be used here or not. That's why I think `WP_DEBUG` works best for this for now.

Nit-pick, can you move it from the extra if clause to the existing one below? IMO no need to have a clause to reset the static cache value when we could instead not use the static cache at all. Having `WP_DEBUG` in the existing clause below is also better in line with how we would like to adjust this code in the future.

Reply:
FWIW, I think none of that is really great, but I think keeping the static and relying on `WP_DEBUG` has the least impact on anything global that we wouldn't be able to change later, and it's relatively close to where we want this code to go eventually.